File: programmers/js/new/level1/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = "";
    const completionMap = new Map();
    completion.forEach((comp) => {
        if (completionMap.has(comp)) {
            completionMap.set(comp, completionMap.get(comp) + 1);
        } else {
            completionMap.set(comp, 1);
        }
    });
    for (let i = 0; i < participant.length; ++i) {
        const part = participant[i];
        if (completionMap.has(part)) {
            completionMap.set(part, completionMap.get(part) - 1);
            if (completionMap.get(part) === 0) {
                completionMap.delete(part);
            }
        } else {
            answer = part;
            break;
        }
    }
    return answer;
}

File: programmers/python/level1/완주하지_못한_선수.py
def solution(participant, completion):
    s = {}
    for c in completion:
        if c in s:
            s[c] += 1
        else:
            s[c] = 1
    for p in participant:
        if p not in s:
            return p
        else:
            s[p] -= 1
            if s[p] == 0:
                del s[p]
    return ''